Src/AutoFixture/EmailAddressLocalPart.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Represents the local part of the email address, defined as everything up to, but not including the @ sign.  
+    /// The requirements enforced on the local part are defined in RFC 3696.  The regular expression for validating 
+    /// the local part borrowed from Phil Haack at 
+    /// http://haacked.com/archive/2007/08/21/i-knew-how-to-validate-an-email-address-until-i.aspx/
```

Comment:
Bingo! :tada: 

Yes, it's unfortunate that the validation used by `MailAddress` is internal, but it also means that for our specific case, it's irrelevant to attempt to match 'the standard' - what we need is to match something that will work with `MailAddress`.

Personally, I don't think it's worthwhile to attempt to reverse-engineer the `MailAddress` validation rules, as they could change (e.g. because of bug fixes). AFAICT, the most pragmatic approach is to attempt to use the string passed into `EmailAddressLocalPart` with a valid domain name ("example.com") in a `try/catch` block. If no exception is thrown, it must be valid.

This will effectively couple `EmailAddressLocalPart` to `MailAddressGenerator`, so this may be one of those fairly rare cases where a nested public class may be appropriate...